### PR TITLE
Allow P Models to accept an array of kphio values

### LINKED ISSUE #261: Allow PModel to accept an array of kphio values.
**Is your feature request related to a problem? Please describe.**

At present, the `kphio` argument to `pyrealm.pmodel.pmodel.PModel`  is expected to be a float value. If this is supplied, it is converted into a `numpy` scalar array, with the value then being used for all calculations. However, users might want to be able to supply per-observation estimates of `kphio` rather than a single global initial value.

**Describe the solution you'd like**
Update `kphio` to accept scalar or array inputs and then check that array `kphio` values broadcast correctly across other array inputs.

### PR DESCRIPTION
**Pull request by davidorme**
# Description

This PR update the `kphio` arguments to `PModel` and `SubdailyPModel` to accept an array of `kphio` estimates. This is largely intended to allow users to test alternative models for calculating `kphio` - there is an approach that accounts for temperature and soil moisture being developed.

The PR then includes a new test suite for this feature that checks that outcomes of passing in an array of kphio values is identical to the predictions of calculating those values using a single value at a time, and that the results are not sensitive to the shape of the inputs. That is repeated for `PModel` and `SubdailyPModel`.

Fixes #261

## Type of change

- [x] New feature (non-breaking change which adds functionality)
- [ ] Optimization (back-end change that speeds up the code)
- [ ] Bug fix (non-breaking change which fixes an issue)

## Key checklist

- [x] Make sure you've run the `pre-commit` checks: `$ pre-commit run -a`
- [x] All tests pass: `$ poetry run pytest`

## Further checks

- [x] Code is commented, particularly in hard-to-understand areas
- [x] Tests added that prove fix is effective or that feature works

### CODE REVIEW

**Review by j-emberton — COMMENTED** [review]
Do we need to add a bit more documentation to guide users on how they can use this new feature? Or is it in line with what we currently have?

**Comment by davidorme** [comment]
> Do we need to add a bit more documentation to guide users on how they can use this new feature? Or is it in line with what we currently have?

Great point - I've added a short paragraph to the user facing PModel docs.

**Review by j-emberton — APPROVED** [review]
Good addition

File: pyrealm/pmodel/pmodel.py [hunk 1]
```diff
@@ -215,11 +217,18 @@ def __init__(
         # Set context specific defaults for kphio to match Stocker paper
         if kphio is None:
             if not self.do_ftemp_kphio:
-                self.init_kphio = 0.049977
+                self.init_kphio = np.array(0.049977)
```
Comment (AmyOctoCat) [inline pyrealm/pmodel/pmodel.py:220]: Could these values be pulled out into a constant rather than using magic numbers?

**Review by AmyOctoCat — APPROVED** [review]
I am not familiar with the theory behind this change, but looks great to me in terms of code from what I can tell. Could possibly set some constants for the magic numbers used.

File: pyrealm/pmodel/pmodel.py (hunk 1 above)
Comment (davidorme) [reply pyrealm/pmodel/pmodel.py:220]: I've got some inbound code that changes the `kphio` part completely, so I _think_ these will disappear in the near future.

**Pull request merged**